src/main/scala/mesosphere/marathon/core/launcher/impl/TaskLabels.scala

```diff
@@ -25,16 +24,17 @@ object TaskLabels {
     val labels = ReservationLabels(resource)
 
     val maybeMatchingFrameworkId = labels.get(FRAMEWORK_ID_LABEL).filter(_ == frameworkId.id)
-    def maybeInstanceId = labels.get(TASK_ID_LABEL).map(Instance.Id.fromReservationId(_))
+    def maybeInstanceId = labels.get(TASK_ID_LABEL).map(Reservation.Id(_).instanceId)
 
     maybeMatchingFrameworkId.flatMap(_ => maybeInstanceId)
   }
 
-  def labelsForTask(frameworkId: FrameworkId, taskId: Task.Id): ReservationLabels = {
+  def labelsForTask(frameworkId: FrameworkId, reservationId: Reservation.Id): ReservationLabels = {
     ReservationLabels(Map(
       FRAMEWORK_ID_LABEL -> frameworkId.id,
       // This uses taskId.reservationId to match against the id that was originally used to create the reservation
-      TASK_ID_LABEL -> taskId.reservationId
+      //TODO(karsten): We probably want to call it RESERVATION_ID_LABEL in the future.
+      TASK_ID_LABEL -> reservationId.label
```

Comment:
well you can at least rename the property 🤷‍♀️ and put this note in there...

Reply:
I don't think I can just rename the property. It belongs to https://jira.mesosphere.com/browse/MARATHON-8517. I'll note it in the JIRA.